pinot-common/src/main/java/com/linkedin/pinot/common/protocols/SegmentCompletionProtocol.java

```diff
@@ -144,17 +146,65 @@ protected String getUri(String msgType) {
         PARAM_INSTANCE_ID + "=" + _instanceId +
           (_reason == null ? "" : ("&" + PARAM_REASON + "=" + _reason));
     }
+
+    public static class Params {
+      private static final String UNKNOWN_SEGMENT = "UNKNOWN_SEGMENT";
+      private static final long UNKNOWN_OFFSET = -1L;
+      private static final String UNKNOWN_INSTANCE = "UNKNOWN_INSTANCE";
+
+      long _offset;
+      String _segmentName;
+      String _instanceId;
+      String _reason;
+
+      public Params() {
+        resetInternal();
+      }
+      public Params reset() {
+        resetInternal();
+        return this;
+      }
+      private void resetInternal() {
+        _offset = UNKNOWN_OFFSET;
+        _segmentName = UNKNOWN_SEGMENT;
+        _instanceId = UNKNOWN_INSTANCE;
+      }
+      public Params setOffset(long offset) {
```

Comment:
Since this is a fluent interface-style class and not a Javabean setter, can you _not_ use the Javabeans names for setters (`setX`) and rather use a name that doesn't overlap with the Javabeans convention.

I skimmed over this code assuming it was standard Javabean property setters and was surprised to see it used later on in a fluent way. I suggest `withX` as the convention for this (eg. `Params.withFoo(1).withBar(2)`) which is readable and doesn't overlap with other conventions.